File: lib/medium/max_sub_array.dart
import "dart:math";

int maxSubArray(List<int> nums) {
  int result = nums[0];
  int ending = nums[0];
  for (int i = 1; i < nums.length; i++) {
    ending = max(ending + nums[i], nums[i]);
    result = max(result, ending);
  }
  return result;
}

File: src/medium/max_sub_array.zig
const std = @import("std");

const expectEqual = std.testing.expectEqual;

pub fn maxSubArray(comptime T: type, nums: []const T) T {
    var result = nums[0];
    var ending = nums[0];
    for (1..nums.len) |index| {
        ending = @max(ending + nums[index], nums[index]);
        result = @max(result, ending);
    }
    return result;
}
